api/versions.py

```
@@ -0,0 +1,95 @@
+from rest_framework.decorators import action
+
+logger = logging.getLogger(__name__)
+
+class VersionViewSet(viewsets.ViewSet):
+    permission_classes = [IsAuthenticated]
+
+    class _RevertSerializer(serializers.Serializer):
+        target_version = serializers.IntegerField(min_value=1)
+
+    @action(detail=False, methods=['post'])
+    def revert(self, request, resource_type: str | None = None, resource_id: str | None = None) -> Response:
```

Ruff (0.12.2)

Missing type annotation for function argument `request`

(ANN001)